### ISSUE TITLE
how to turn off not letting the tasks timeline turn on by default when I start obsidian? 

### ISSUE THREAD

**Issue by Signets** [Leonezz/obsidian-tasks-calendar-wrapper#70]
I'm using this plugin Tasks Calendar Wrapper in obsidian, how do I turn off not letting the tasks timeline turn on by default when I start obsidian? Since there's a shortcut, I think I'd like to use the shortcut to bring it up when I need it, without him having a strong reminder on the right side. It's annoying to see unfinished tasks right at startup, and it feels like I'm being led around by tasks, and it feels bad that I don't have a sense of dominating my life.
我在obsidian中使用了这款插件Tasks Calendar Wrapper，如何关闭不让tasks timeline在启动obsidian时默认开启？既然有快捷键，我想我需要的时候用快捷键调出它，而不需要他在右侧强提醒。启动时就看到未完成的任务很烦，觉得自己被任务带着走，感觉很糟糕，没有主导生活的感觉。

**Comment by Leonezz** [comment]
I am sorry you are not feeling good using this plugin, I will see what I can do about this feature.

**Comment by ksdavidc** [comment]
Not causing so much stress, but there does seem to be a button there to not enable focus on today at startup Is that the same thing? If so, it does not seem to work. Annoying enough to possibly make me disable the plugin until a workaround is available.

**Comment by Leonezz** [comment]
Hi @ksdavidc , Do you mean the Today Focus On Load option is not working? Could you please provide more infomation cuz I cannot reproduce it in my environment.
![image](https://github.com/Leonezz/obsidian-tasks-calendar-wrapper/assets/33564074/58139c23-1c85-4aae-934b-5d0c440f5327)

Also if the default behavior (i.e. shows up on obsidian opening) is too much disturbing, I will add an option for it @Signets .

**Comment by ksdavidc** [comment]
HI, sorry for the delay in responding. I just noticed it. 


> Hi @ksdavidc , Do you mean the Today Focus On Load option is not working? Could you please provide more infomation cuz I cannot reproduce it in my environment. ![image](https://private-user-images.githubusercontent.com/33564074/286166498-58139c23-1c85-4aae-934b-5d0c440f5327.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNTM3MDUsIm5iZiI6MTcwMzE1MzQwNSwicGF0aCI6Ii8zMzU2NDA3NC8yODYxNjY0OTgtNTgxMzljMjMtMWM4NS00YWFlLTkzNGItNWQwYzQ0MGY1MzI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDEwMTAwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjMWY4MzA2YThkMmJkMTc3ODA5NzUzYjc1NzA4MDJhZWM4MmRjMDU0YjY0MzY4ZDhmNzRkM2M0NGE0OWQ1MmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.b06ZDX7Pcia0MQqBZHNrYnNhs35nnKdKK0Slas8HNoM)
> 

I think so, but I am not sure if that means what I think it means. If that option is intended to control having the plugin page appear in the right sidebar when opening Obsidian, then yes, it is not working, or rather it is working too well. It does open on load, but I seem to be unable to STOP the sidebar from showing on open, even if I turn off Today focus on load.

> Also if the default behavior (i.e. shows up on obsidian opening) is too much disturbing, I will add an option for it @Signets .

I too would like to have a switch to turn off the default. I like this plugin but I prefer to see my regular calendar when I open so I have had to disable it for now...

**Comment by ksdavidc** [comment]
Slight nudge. 
Attached virgin sample vault exhibiting the behavior.

focus on load turned off
<img width="979" alt="Screenshot 2024-03-10 at 6 38 12 AM" src="https://github.com/Leonezz/obsidian-tasks-calendar-wrapper/assets/62230004/6635f193-ab8e-49a1-9288-294b17279bbc">
different tab in sidebar before closing

<img width="991" alt="Screenshot 2024-03-10 at 6 38 27 AM" src="https://github.com/Leonezz/obsidian-tasks-calendar-wrapper/assets/62230004/92ed8ef4-8d0c-4729-91b0-0fadc9302bd6">
closed and reopened. Focus showing in sidebar. Please, no.
<img width="997" alt="Screenshot 2024-03-10 at 6 38 45 AM" src="https://github.com/Leonezz/obsidian-tasks-calendar-wrapper/assets/62230004/261f5f45-0474-4ce4-8410-0606f4435fe0">

zipped vault.
[test.zip](https://github.com/Leonezz/obsidian-tasks-calendar-wrapper/files/14547866/test.zip)

**Comment by ksdavidc** [comment]
I had to fix it myself. I changed the following function:

```
 async activateView(type) {
    if (type !== TIMELINE_VIEW) {
      return;
    }
    this.app.workspace.detachLeavesOfType(type);
    try {
      await this.app.workspace.getRightLeaf(false).setViewState({
        type,
        active: false // change this to false to prevent load on start
      });
      this.app.workspace.revealLeaf(
        this.app.workspace.getLeavesOfType(type).first()
      );
    } catch (e) {
      console.log(e);
    }
  }
};
```

**Comment by Leonezz** [comment]
@ksdavidc Sorry I could not get it fixed in time, I was busy preparing to defend my thesis. A pr would be great if you have fixed it in your own codebase.

**Comment by ksdavidc** [comment]
> @ksdavidc Sorry I could not get it fixed in time, I was busy preparing to defend my thesis. A pr would be great if you have fixed it in your own codebase.

When I said "fix", my tongue was in my cheek. What I did was break it. It was an awful hack, as I have no clue what I am doing. I stumbled on figuring out that making the setViewState() function inoperative seemed to work, but I have no idea what other consequences this might have elsewhere, or even if the effects are permanent or wear off. I only changed that one commented statement from "true" to false, so there is nothing much to pr.
